cardano-submit-api/src/Cardano/TxSubmit/Types.hs

```diff
@@ -12,36 +16,39 @@ module Cardano.TxSubmit.Types
   , EnvSocketError(..)
   , TxCmdError(..)
   , RawCborDecodeError(..)
-  , renderTxSubmitWebApiError
   , renderTxCmdError
   ) where
 
-import           Cardano.Api (Error (..), TxId, textShow)
+import           Cardano.Api (AnyCardanoEra, Error (..), TxId, TxValidationErrorInCardanoMode (..),
+                   textShow)
 import           Cardano.Api.Pretty
 import           Cardano.Binary (DecoderError)
-import           Data.Aeson (ToJSON (..), Value (..))
+import           Cardano.TxSubmit.Orphans ()
+import           Data.Aeson (ToJSON (..), Value (..), (.=))
+import qualified Data.Aeson as Aeson
 import           Data.ByteString.Char8 (ByteString)
 import           Data.Text (Text)
-import           Formatting (build, sformat)
 import           GHC.Generics (Generic)
 import           Network.HTTP.Media ((//))
-import           Ouroboros.Consensus.Cardano.Block (EraMismatch (..))
 import           Servant (Accept (..), JSON, MimeRender (..), MimeUnrender (..), PostAccepted,
                    ReqBody, (:>))
 import           Servant.API.Generic (ToServantApi, (:-))
 
 import qualified Data.ByteString.Lazy.Char8 as LBS
+import qualified Data.Text as T
 
 newtype TxSubmitPort = TxSubmitPort Int
 
 -- | The errors that the raw CBOR transaction parsing\/decoding functions can return.
 --
 newtype RawCborDecodeError = RawCborDecodeError [DecoderError]
-  deriving (Eq, Show)
+  deriving (Eq, Generic, Show)
 
 instance Error RawCborDecodeError where
   prettyError (RawCborDecodeError decodeErrors) = "RawCborDecodeError decode error: " <> pshow (fmap pshow decodeErrors)
```

Comment:
I think this instance can be simplified a bit
```suggestion
  prettyError (RawCborDecodeError decodeErrors) = "RawCborDecodeError decode error:" <+> pshow decodeErrors
```

Reply:
This instance wasn't added in the PR.